src/wallet/wallet.cpp

```diff
@@ -4354,7 +4359,10 @@ bool DoMigration(CWallet& wallet, WalletContext& context, bilingual_str& error,
 
                 // Add to the wallet
                 WalletDescriptor w_desc(std::move(descs.at(0)), creation_time, 0, 0, 0);
-                data->watchonly_wallet->AddWalletDescriptor(w_desc, keys, "", false);
+                if(auto res = data->watchonly_wallet->AddWalletDescriptor(w_desc, keys, "", false); !res)
```

Comment:
The `res` variable is already used in this function, by declaring it here again you are shadowing the other one. Would be better to rename it to `res_spkm` or `spkm_res`.

Also, missing whitespace after `if`.